From bc3d79f0ed379900618598a9a27f1dbf7bab8942 Mon Sep 17 00:00:00 2001
From: WeirdTreeThing <bradyn127@protonmail.com>
Date: Fri, 27 Jun 2025 09:27:35 -0400
Subject: [PATCH] arm64: dts: exynos: gs101-bluejay: add new board file

Bluejay is Google's code name for Pixel 6a. Similar to Pixel 6
(Oriole), this is also based around its Tensor gs101 SoC.
---
 arch/arm64/boot/dts/exynos/google/Makefile    |  3 +-
 .../boot/dts/exynos/google/gs101-bluejay.dts  | 28 +++++++++++++++++++
 2 files changed, 30 insertions(+), 1 deletion(-)
 create mode 100644 arch/arm64/boot/dts/exynos/google/gs101-bluejay.dts

diff --git a/arch/arm64/boot/dts/exynos/google/Makefile b/arch/arm64/boot/dts/exynos/google/Makefile
index 7385f82b03c9..93c82490820c 100644
--- a/arch/arm64/boot/dts/exynos/google/Makefile
+++ b/arch/arm64/boot/dts/exynos/google/Makefile
@@ -2,4 +2,5 @@
 
 dtb-$(CONFIG_ARCH_EXYNOS) += \
 	gs101-oriole.dtb \
-	gs101-raven.dtb
+	gs101-raven.dtb \
+	gs101-bluejay.dtb
diff --git a/arch/arm64/boot/dts/exynos/google/gs101-bluejay.dts b/arch/arm64/boot/dts/exynos/google/gs101-bluejay.dts
new file mode 100644
index 000000000000..a7e162cdaeee
--- /dev/null
+++ b/arch/arm64/boot/dts/exynos/google/gs101-bluejay.dts
@@ -0,0 +1,28 @@
+// SPDX-License-Identifier: GPL-2.0-only
+/*
+ * Bluejay Device Tree
+ *
+ * Copyright 2021-2023 Google LLC
+ */
+
+/dts-v1/;
+
+#include "gs101-pixel-common.dtsi"
+
+/ {
+	model = "Bluejay";
+	compatible = "google,gs101-bluejay", "google,gs101";
+};
+
+&cont_splash_mem {
+	reg = <0x0 0xfac00000 (1080 * 2400 * 4)>;
+	status = "okay";
+};
+
+&framebuffer0 {
+	width = <1080>;
+	height = <2400>;
+	stride = <(1080 * 4)>;
+	format = "a8r8g8b8";
+	status = "okay";
+};
-- 
2.49.0

